src/stub/scripts/xstrip.py

```diff
@@ -161,7 +161,8 @@ def do_file(fn):
         # ELF64 LE
         eh, idata = idata[:64], idata[64:]
         e_shnum, e_shstrndx = struct.unpack("<HH", eh[60:64])
-        assert e_shstrndx + 3 == e_shnum
+        if ( e_shstrndx + 1 != e_shnum  and  e_shstrndx + 3 != e_shnum ):
+            assert e_shnum == e_shstrndx
```

Comment:
On ubuntu 16.04 I get 
python ./../../src/stub/scripts/xstrip.py --with-dump=tmp/powerpc64le-darwin.dylib-entry.bin.dump --create-bindump=tmp/powerpc64le-darwin.dylib-entry.bin.bindump tmp/powerpc64le-darwin.dylib-entry.bin
e_shnum= 22 e_shstrndx = 19
but on Ubuntu 17.04, debian and fedora I have 
e_shnum= 22 e_shstrndx = 21

I agree it must be a separated commit, but I need to understand what it makes that difference. I have checked it is not related to gcc version.